algopy_testing/src/algopy_testing/arc4.py

```diff
@@ -50,9 +49,61 @@ def from_log(cls, log: algopy.Bytes, /) -> typing.Self:
         ...
 
 
+class String(_ABIEncoded):
+    """An ARC4 sequence of bytes containing a UTF8 string"""
+
+    _value: bytes
+
+    def __init__(self, value: algopy.String | str = "", /) -> None:
+        bytes_value = (
+            value.bytes.value
+            if isinstance(value, algopy.String)
+            else as_string(value).encode("utf-8")
```

Comment:
Unfortunately can't use as_string here either as it allows broader types than the stubs indicate.

Can use a `match` to handle `str` and `algopy.String` and then the default case can just be a type error

Reply:
It has been fixed in this commit 9e8f1dfaa2b440b9ef3241ad3290abf7df87bce9.